### PR TITLE
Feature/websockets

### PR DESCRIPTION
**Pull request by clementgerard1**
add /timeside/api/users/ GET alias : "me" username determine user with token value
TODO : remove possibility to create new user with username "me" (not trivial because a custom User Model is needed)

Minimal websockets implementation with messages for events :
- task started
- experience started
- experience failed
- experience done
- experience completion

Websockets authentification use Django REST Framework token

Completion is triggered in decoder aubio or file (not tested with file) with COMPLETION_INTERVAL Django setting which determine the number of block between each message

For testing in dev mode : go to localhost:13000, enter token and see messages in dev console when you trigger analysis or experiences

Just tested with /timeside/api/analysis_tracks/ POST request
Need to be tested in all situations

### CODE REVIEW

**Comment by yomguy** [comment]
Wonderfull job @clementgerard1 thanks! I will start a short review.

**Comment by yomguy** [comment]
hi @clementgerard1 the django and the node parts are OK but the core part should not include any links to them. The related usecase is that someone should be able to run a process through a python shell without any worry about the server side. Sorry if this was not clear and detailed in the request.

Maybe we should rather try to retrieve the state of the decoder injecting a callback into the [pipe.run()](https://github.com/Parisson/TimeSide/blob/dev/timeside/server/models.py#L549) and use the [task.run] method to push the value into the socket. This will ensure that the decoder can run without knowing the server and let the backend do the web notification job. 

cc/ @rherbelleau @gnuletik

**Review by gnuletik — COMMENTED** [review]
Thanks for this great PR @clementgerard1 

I just have some minor stuff : 

The `CACHE_RESULT` implementation seems a bit hacky.
I think the proper way to handle this would be to provide a route parameter (createTask or createAnalysisTrack).
What's your thoughts on this @yomguy ?

Could you serve the index.html page in dev only ?

**Comment by gnuletik** [comment]
Also, if we are extending JS usage in the team, would you like to choose one JS coding standard like :
- https://github.com/standard/eslint-config-standard
- https://github.com/airbnb/javascript

**Comment by clementgerard1** [comment]
> hi @clementgerard1 the django and the node parts are OK but the core part should not include any links to them. The related usecase is that someone should be able to run a process through a python shell without any worry about the server side. Sorry if this was not clear and detailed in the request.
> 
> Maybe we could rather try to retrieve the state of the decoder injecting a callback into the [pipe.run](https://github.com/Parisson/TimeSide/blob/dev/timeside/server/models.py#L549) method and use the [task.run](https://github.com/Parisson/TimeSide/blob/dev/timeside/server/tasks.py#L19) one to push the value into the socket. This will ensure that the decoder can run without knowing the server and let the backend do the web notification job.
> 
> cc/ @rherbelleau @gnuletik

Done with progress_callback function defined in run method of Item model in server/models.py

File: env/debug.yml [hunk 1]
```diff
@@ -32,3 +32,16 @@ services:
...
+  websockets:  
+    build: ./websockets
+    ports:
+      - "9031:8000"
+    environment:
+      - ORIGIN=http://localhost:9030
+    command: "npm run dev"
+    volumes:
+      - ./websockets:/srv/app/
+    user: "root:root"
```
[the remark this answers is not in the document]
Comment (clementgerard1) [reply env/debug.yml:44]: I don't remember why I did this on another project but I think it was because the volume sharing was complicated without it

**Comment by clementgerard1** [comment]
> Also, if we are extending JS usage in the team, would you like to choose one JS coding standard like :
> 
> * https://github.com/standard/eslint-config-standard
> * https://github.com/airbnb/javascript

Yes, what did you use in other js POW projects ?

File: websockets/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "install:build:prod": "npm install && npm run build && npm run prod",
+    "lint": "eslint server.js --ext .js"
+  },
+  "author": "Clément Gérard",
+  "license": "ISC",
+  "dependencies": {
+    "@babel/cli": "^7.13.0",
+    "@babel/core": "^7.12.16",
+    "@babel/eslint-parser": "^7.12.16",
+    "@babel/node": "^7.12.16",
```
[the remark this answers is not in the document]
Comment (clementgerard1) [reply websockets/package.json:21]: I update node server with ES6 features

**Comment by clementgerard1** [comment]
> 
> Could you serve the index.html page in dev only ?
index.ejs page is only served when DEBUG setting is on

**Comment by clementgerard1** [comment]
> > Also, if we are extending JS usage in the team, would you like to choose one JS coding standard like :
> > 
> > * https://github.com/standard/eslint-config-standard
> > * https://github.com/airbnb/javascript
> 
> Yes, what did you use in other js POW projects ?

I add standard config and rewrite for linter acceptation (https://github.com/standard/eslint-config-standard)

**Comment by yomguy** [comment]
Wonderful job @clementgerard1 ! 
We will just need to fix the user running nodejs.

**Pull request merged**